### PR TITLE
Compiles with nim 0.19.0 and passes tests (some of the generated code…

### PR DESCRIPTION
**Pull request by jacereda**
… doesn't compile though)

### CODE REVIEW

File: py2nim.nimble [hunk 1]
```diff
@@ -17,5 +17,4 @@ proc configForTests() =
 
 task test, "run CPU tests":
   configForTests()
-  setCommand "c", "tests/all.nim"
-
+  setCommand "c", "tests/file_test.nim"
```
[the remark this answers is not in the document]
Comment (jacereda) [reply py2nim.nimble]: Probably just me missing something... what's all.nim? Is that generated somehow? I didn't find it under tests/...

File: py2nim_helpers.nim [hunk 2]
```diff
@@ -1,31 +1,49 @@
...
+    let (rk, rv) = op(it)
+    result.add(rk, rv)
+
+template mapTable*[A, B](t: TableRef[A, B], op: untyped): untyped =
+  mapTableTP(t, proc(k: A, v: B): auto = (let k{.inject.} = k; let v{.inject.} = v; op))
+
+template mapTable*[A](t: seq[A], op: untyped): untyped =
+  mapTableSP(t, proc(it: A): auto = (let it{.inject.} = it; op))
+
+# macro with*(a: untyped, b: untyped): untyped =
```
[the remark this answers is not in the document]
Comment (jacereda) [reply py2nim_helpers.nim]: Oops, I'll get them back...

File: py2nim_helpers.nim [hunk 3]
```diff
@@ -1,31 +1,49 @@
...
+    let (rk, rv) = op(k, v)
+    result.add(rk, rv)
+
+proc mapTableSP[A, B, C](t: seq[A], op: proc(it: A): (B, C)): TableRef[B, C] =
+  result = newTable[B, C](rightSize(t.len))
+  for it in t.items:
+    let (rk, rv) = op(it)
+    result.add(rk, rv)
+
+template mapTable*[A, B](t: TableRef[A, B], op: untyped): untyped =
```
[the remark this answers is not in the document]
Comment (jacereda) [reply py2nim_helpers.nim]: Yeah, as you say this can be changed. In any case, IIRC, with the right combination of flags it will just inline it.

**Comment by jacereda** [comment]
I have more changes regarding generating tuples for heterogeneous lists. Do you prefer that in a separate PR?

File: py2nim.nimble (hunk 1 above)
[the remark this answers is not in the document]
Comment (alehander42) [reply py2nim.nimble]: ah, you're right

[the remark this answers is not in the document]
Comment (alehander42) [reply py2nim.nimble]: sorry haha

File: py2nim_helpers.nim (hunk 3 above)
[the remark this answers is not in the document]
Comment (alehander42) [reply py2nim_helpers.nim]: I just wonder if mapTable or the comprehension DSL are a better translation: the comprehensions look a bit more like python, but mapTable is more like sequtils. Anyway, we can change this in the future, we already had mapTable planned

**Comment by alehander42** [comment]
No, it's ok. Thank you for the fixes and additions!

**Pull request merged**